research/src/pages/popup/popup.proposal.mdx

```diff
@@ -0,0 +1,434 @@
+   [showing a `popup` element steps](#showing-popup-steps) with _candidate subject_ set to the current
+   `popup` element.
+2. Otherwise, return.
+
+<p class="note">
+  As a result of following these run steps: in the event that initiallyopen is specified on multiple
+  popups, the last of these popups (with the initiallyopen attribute) in the DOM order will be the
+  popup that is rendered upon document load.
+</p>
+
+### The `show()` method
+
+The `show()` method renders the `popup` element, such that the user can interact with it.
+
+**Example:**
+
+```html
+<popup id="newFeatureUI">
+  <p>
+    <strong>New!</strong>
+    I’m some sort of educational UI…
+  </p>
+</popup>
+
+<script>
+  if (upsellNewFeature) {
+    document.getElementById('newFeatureUI').show()
+  }
+</script>
+```
+
+The show() method steps are to run the [showing a `popup` element steps](#showing-popup-steps) with
+[this](https://heycam.github.io/webidl/#this) with _candidate subject_ set to the new candidate
+`popup` element to be shown.
+
+### The `popup` attribute
+
+The `popup` attribute indicates that a related `popup` element should be shown or hidden as the
+user interacts with the element where the `popup` attribute is specified. If the attribute is
+specified, the attribute's value must be the ID of a `popup` element in the same
+[node document](https://dom.spec.whatwg.org/#concept-node-document).
+
+**Example:** Invoking the `button` element in this example should show/hide the `popup` element,
+depending on whether or not the `popup` element is currently shown:
+
+```html
+<button id="menuButton" popup="menuPopup">Menu</button>
+<popup id="menuPopup" role="menu" anchor="menuButton">
+  <!-- Markup for menuitems goes here -->
+</popup>
+```
+
+The `popup` attribute is supported on a subset of
+[interactive elements](https://html.spec.whatwg.org/multipage/interactive-elements.html):
+
+- `button`
+- `input` in the `button` state
+- `input` in the `email`, `number`, `search`, `tel`, `text`, or `url` states
+
+#### Showing/hiding a `popup` element via the `popup` attribute
+
+When an element with the `popup` attribute specified receives a user interaction that should show
+the related `popup` element, this element shall be known as the `popup` element's _invoker_.
+
+<p class="note">
+  Multiple elements may have a popup attribute that refers to the same popup element. A given popup
+  element only has 0 or 1 invokers at a given time. For example, if the popup element was shown
+  programmatically, the invoker is null. Otherwise, the element that the user actually interacted
+  with to show the popup is its sole invoker.
+</p>
+
+If the _invoker_ is a `button` element or an `input` element in the `button` state, and the
+_invoker_ is activated; OR the _invoker_ is an input in the `email`, `number`, `search`, `tel`,
+`text`, or `url` states and focus is set to the _invoker_, the user agent must run the following
+steps:
```

Comment:
Sounds good. Couple things I'm not sure of:

1. We'd have to run these steps from 8 different places (element types + applicable states) across the HTML spec. Should the steps live in one central location that we can call out to from each applicable `button`/`input` state? Or do we prefer to repeat everywhere they should appear?
2. Just a heads up that I think it would be good to have some note here pointing to the various different places that we move these steps, so that implementers/authors/other audiences can cross-reference from `popup`. Happy to take suggestions on how that's done.

Also, I don't think there's any effect on the focus update steps from this piece directly, since we are persisting user focus (unless moved via the `autofocus` or `delegatesfocus` attribute).